### PR TITLE
Adding .net samples for batch inferencing

### PR DESCRIPTION
**Pull request by mastloui-msft**
# PR into Azure/azureml-examples

## Checklist

I have:

- [ ] read and followed the contributing guidelines

## Changes

-

fixes #

### CODE REVIEW

File: samples/.net/endpoints/batch/Clients/EndpointClient.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                    InputData = new Dictionary<string, BatchScoringInput>
+                    {
+                        {
+                            "myInput",
+                            new BatchScoringInput
+                            {
+                                JobInputType = JobInputType.UriFolder,
+                                Uri = inputFolderUri,
+                            }
+                        },
```
Comment (Iyer-Narayan) [inline samples/.net/endpoints/batch/Clients/EndpointClient.cs:44]: Could you use the following syntax, which helps reduce the number of curly braces and their indentations?
```
["myInput"] = new BatchScoringInput
{
    ...
}
```

Comment (Iyer-Narayan) [reply samples/.net/endpoints/batch/Clients/EndpointClient.cs:44]: Alternatively creating a helper method would further reduce the noise: 
```
    BatchScoringPayload payload = new BatchScoringPayload
    {
        Properties = new BatchScoringProperties
        {
            InputData = CreateDictionary(
                "myInput",
                new BatchScoringInput
                {
                    JobInputType = JobInputType.UriFolder,
                    Uri = inputFolderUri,
                }),
            OutputData = CreateDictionary(
                "myOutput",
                new BatchScoringOutput
                {
                    JobOutputType = JobOutputType.UriFile,
                    Uri = outputFileUri
                })
        },
    };

    private static IDictionary<string, T> CreateDictionary<T>(string key, T value)
        => new Dictionary<string, T> { [key] = value };
        
```

**Review by Iyer-Narayan — CHANGES_REQUESTED** [review]

File: samples/.net/endpoints/batch/Contracts/BatchScoringProperties.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+﻿// <copyright file="BatchScoringProperties.cs" company="Microsoft">
+// Copyright (c) Microsoft. All rights reserved.
+// </copyright>
+
+namespace BatchInferencingSamples
+{
+    internal class BatchScoringProperties
+    {
+        /// <summary>
+        /// Settings of the source data.
```
Comment (Iyer-Narayan) [inline samples/.net/endpoints/batch/Contracts/BatchScoringProperties.cs:10]: Could we use standard comment style "Gets or sets..." - this clearly tells people teh nature of the property (gettable, settable) by seeing the intellisense.

File: samples/.net/endpoints/batch/Program.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    internal class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            TokenCredential credentials = new DefaultAzureCredential();
+            
+            Guid subscriptionId = Guid.Parse("<your_subscription_id>");
+            string resourceGroupName = "<your_resource_group_name>";
+            string workspaceName = "<your_workspace_name>";
+            string batchEndpointName = "<your_endpoint_name>";
```
Comment (Iyer-Narayan) [inline samples/.net/endpoints/batch/Program.cs:20]: Should we parse args for these?

**Review by Iyer-Narayan — CHANGES_REQUESTED** [review]

File: samples/.net/endpoints/batch/Clients/EndpointClient.cs (hunk 1 above)
Comment (mastloui-msft) [reply samples/.net/endpoints/batch/Clients/EndpointClient.cs:44]: I opted for the first suggestion!

File: samples/.net/endpoints/batch/Program.cs (hunk 3 above)
Comment (mastloui-msft) [reply samples/.net/endpoints/batch/Program.cs:20]: hmmm we can iterate on the args parsing..!

**Review by Iyer-Narayan — APPROVED** [review]

File: samples/.net/endpoints/batch/Clients/AuthenticationClient.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        /// Gets an access token to score.
+        /// </summary>
+        /// <param name="credentials">The user's credentials.</param>
+        /// <returns>A string representation of the access token.</returns>
+        public async Task<string> GetAccessToken(TokenCredential credentials)
+        {
+            // Given this is a dataplane operation we need the authentication scope to come from Azure Machine Learning
+            string[] scopes = new string[]
+            {
+                "https://ml.azure.com/.default"
```
Comment (chupadhy) [inline samples/.net/endpoints/batch/Clients/AuthenticationClient.cs:21]: What is the default? Isn't the scope just ml.azure.com?

**Review by chupadhy — APPROVED** [review]

File: samples/.net/endpoints/batch/Clients/AuthenticationClient.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+namespace BatchInferencingSamples
+{
+    internal class AuthenticationClient
+    {
+        /// <summary>
+        /// Gets an access token to score.
+        /// </summary>
+        /// <param name="credentials">The user's credentials.</param>
+        /// <returns>A string representation of the access token.</returns>
+        public async Task<string> GetAccessToken(TokenCredential credentials)
```
Comment (Iyer-Narayan) [inline samples/.net/endpoints/batch/Clients/AuthenticationClient.cs:16]: Could you add Async suffix to async methods?

**Review by Iyer-Narayan — CHANGES_REQUESTED** [review]

File: samples/.net/endpoints/batch/Contracts/BatchScoringOutput.cs [hunk 6]
```diff
@@ -0,0 +1,19 @@
+﻿// <copyright file="BatchScoringOutput.cs" company="Microsoft">
+// Copyright (c) Microsoft. All rights reserved.
+// </copyright>
+
+namespace BatchInferencingSamples
```
Comment (Iyer-Narayan) [inline samples/.net/endpoints/batch/Contracts/BatchScoringOutput.cs:5]: How do you like `Microsoft.Azure.MachineLearning.Samples.BatchInferencing`?

**Review by Iyer-Narayan — CHANGES_REQUESTED** [review]

File: samples/.net/endpoints/batch/Clients/AuthenticationClient.cs (hunk 4 above)
Comment (mastloui-msft) [reply samples/.net/endpoints/batch/Clients/AuthenticationClient.cs:21]: It's the recommended way in c#!

**Review by Iyer-Narayan — APPROVED** [review]

**Pull request merged**